contracts/staking/contracts/src/staking_pools/MixinStakingPoolMakers.sol

```diff
@@ -0,0 +1,210 @@
+            confirmed: false
+        });
+        poolJoinedByMakerAddress[makerAddress] = poolJoinStatus;
+        poolById[poolId].numberOfMakers = uint256(poolById[poolId].numberOfMakers).safeSub(1).downcastToUint32();
+
+        // Maker has been removed from the pool`
+        emit MakerRemovedFromStakingPool(
+            poolId,
+            makerAddress
+        );
+    }
+
+    /// @dev Returns the pool id of the input maker.
+    /// @param makerAddress Address of maker
+    /// @return Pool id, nil if maker is not yet assigned to a pool.
+    function getStakingPoolIdOfMaker(address makerAddress)
+        public
+        view
+        returns (bytes32)
+    {
+        if (isMakerAssignedToStakingPool(makerAddress)) {
+            return poolJoinedByMakerAddress[makerAddress].poolId;
+        } else {
+            return NIL_POOL_ID;
+        }
+    }
+
+    /// @dev Returns true iff the maker is assigned to a staking pool.
+    /// @param makerAddress Address of maker
+    /// @return True iff assigned.
+    function isMakerAssignedToStakingPool(address makerAddress)
+        public
+        view
+        returns (bool)
+    {
+        return poolJoinedByMakerAddress[makerAddress].confirmed;
+    }
+
+    /// @dev Returns the current number of makers in a given pool.
+    /// @param poolId Unique id of pool.
+    /// @return Size of pool.
+    function getNumberOfMakersInStakingPool(bytes32 poolId)
+        public
+        view
+        returns (uint256)
+    {
+        return poolById[poolId].numberOfMakers;
+    }
+
+    /// @dev Adds a maker to a staking pool. Note that this is only callable by the pool operator.
+    /// Note also that the maker must have previously called joinStakingPoolAsMaker.
+    /// @param poolId Unique id of pool.
+    /// @param makerAddress Address of maker.
+    function _addMakerToStakingPool(
+        bytes32 poolId,
+        address makerAddress
+    )
+        internal
+    {
+        // Is the maker already in a pool?
+        if (isMakerAssignedToStakingPool(makerAddress)) {
+            LibRichErrors.rrevert(LibStakingRichErrors.MakerPoolAssignmentError(
+                LibStakingRichErrors.MakerPoolAssignmentErrorCodes.MakerAddressAlreadyRegistered,
+                makerAddress,
+                getStakingPoolIdOfMaker(makerAddress)
+            ));
+        }
+
+        // Is the maker trying to join this pool; or are they the operator?
+        bytes32 makerPendingPoolId = poolJoinedByMakerAddress[makerAddress].poolId;
+        if (makerPendingPoolId != poolId && makerAddress != poolById[poolId].operator) {
+            LibRichErrors.rrevert(LibStakingRichErrors.MakerPoolAssignmentError(
+                LibStakingRichErrors.MakerPoolAssignmentErrorCodes.MakerAddressNotPendingAdd,
+                makerAddress,
+                makerPendingPoolId
+            ));
+        }
+
+        // Is the pool already full?
+        if (getNumberOfMakersInStakingPool(poolId) == maximumMakersInPool) {
```

Comment:
We already reference `poolById[poolId]` above, so this call is using an extra sload. We can load the pool in memory and reference that instead.